lib/spack/spack/builder.py

```diff
@@ -581,5 +581,9 @@ def __str__(self):
 
 
 # Export these names as standalone to be used in packages
-run_after = PhaseCallbacksMeta.run_after
-run_before = PhaseCallbacksMeta.run_before
+def run_after(phase: str, when: Optional[str] = None):
```

Comment:
Is there a good reason to replace a direct call to the `run_after` method with a wrapper around it?

Reply:
I'm using the wrapper function as a means to add type annotations, without them MyPy errors with:
```
lib/spack/spack/build_systems/_checks.py:128: error: Cannot determine type of "run_after"  [has-type]
```

The alternative is marking it with some long `Callable[...]` type. Would that be preferred?

Reply:
New patch guards these wrapper functions under `if TYPE_CHECKING:`, so they don't cause any slowdown at runtime.